tests/CppUTest/JUnitOutputTest.cpp

```diff
@@ -162,7 +162,7 @@ class JUnitTestOutputTestRunner
 
 public:
 
-    JUnitTestOutputTestRunner(TestResult result) :
+    explicit JUnitTestOutputTestRunner(const TestResult& result) :
```

Comment:
Please remove the "explicit"

Reply:
why? is implicit conversion expected? no tests failed (my static analyzer complaints about this pattern)

Reply:
By the way, what about TestResult: I see it to be a pattern to pass it by value. Is it OK that I changed it to cost reference?
(As I'm now touching most of test outputs to handle returning error for 0 run tests, I'm just fixing those where I touch the file. But I can stop.)

Reply:
re: explicit. Actually, you can keep it. I thought the explicit keyword wasn't used anywhere, but I am wrong.

Reply:
re: TestResult. Since a TestResult is just a simple value object, there is no real benefit for passing it as reference. It is not needed... but also not really needed to be removed as... it isn't very important.

Reply:
Test result: I would call this "simple value object":
```cxx
private:
    TestOutput& output_;
    int testCount_;
    int runCount_;
    int checkCount_;
    int failureCount_;
    int filteredOutCount_;
    int ignoredCount_;
    long totalExecutionTime_;
    long timeStarted_;
    long currentTestTimeStarted_;
    long currentTestTotalExecutionTime_;
    long currentGroupTimeStarted_;
    long currentGroupTotalExecutionTime_;
```
I.e. copying so many bytes is questionable. But I agree that probably the compiler is clever enough to remove the copy.


Bottom line: I'll keep this line as as and only fix the other one regarding function extraction